src/ParallelAlgorithms/Events/ObserveFields.hpp

```diff
@@ -199,35 +233,38 @@ class ObserveFields<VolumeDim, ObservationValueTag, tmpl::list<Tensors...>,
             NonSolutionTensors::type::size()...},
         0_st));
 
-    const auto record_tensor_components = [ this, &components, &element_name ](
-        const auto tensor_tag_v, const auto& tensor) noexcept {
-      using tensor_tag = tmpl::type_from<decltype(tensor_tag_v)>;
-      if (variables_to_observe_.count(db::tag_name<tensor_tag>()) == 1) {
-        for (size_t i = 0; i < tensor.size(); ++i) {
-          components.emplace_back(element_name + db::tag_name<tensor_tag>() +
-                                      tensor.component_suffix(i),
-                                  tensor[i]);
-        }
-      }
-    };
+    const auto record_tensor_components =
+        [this, &components, &element_name, &interp_matrices, &mesh](
+            const auto tensor_tag_v, const auto& tensor) noexcept {
+          using tensor_tag = tmpl::type_from<decltype(tensor_tag_v)>;
+          if (variables_to_observe_.count(db::tag_name<tensor_tag>()) == 1) {
+            for (size_t i = 0; i < tensor.size(); ++i) {
+              components.emplace_back(
+                  element_name + db::tag_name<tensor_tag>() +
+                      tensor.component_suffix(i),
+                  apply_matrices(interp_matrices, tensor[i], mesh.extents()));
+            }
+          }
+        };
     record_tensor_components(tmpl::type_<coordinates_tag>{},
                              inertial_coordinates);
     EXPAND_PACK_LEFT_TO_RIGHT(record_tensor_components(
         tmpl::type_<AnalyticSolutionTensors>{}, analytic_solution_tensors));
     EXPAND_PACK_LEFT_TO_RIGHT(record_tensor_components(
         tmpl::type_<NonSolutionTensors>{}, non_solution_tensors));
 
-    const auto record_errors = [ this, &components, &element_name ](
-        const auto tensor_tag_v, const auto& tensor,
-        const auto& analytic_tensor) noexcept {
+    const auto record_errors = [this, &components, &element_name,
+                                &interp_matrices, &mesh](
+                                   const auto tensor_tag_v, const auto& tensor,
+                                   const auto& analytic_tensor) noexcept {
       using tensor_tag = tmpl::type_from<decltype(tensor_tag_v)>;
       if (variables_to_observe_.count(db::tag_name<tensor_tag>()) == 1) {
         for (size_t i = 0; i < tensor.size(); ++i) {
           DataVector error = tensor[i] - analytic_tensor[i];
-          components.emplace_back(element_name + "Error(" +
-                                      db::tag_name<tensor_tag>() + ")" +
-                                      tensor.component_suffix(i),
-                                  std::move(error));
+          components.emplace_back(
+              element_name + "Error(" + db::tag_name<tensor_tag>() + ")" +
+                  tensor.component_suffix(i),
+              apply_matrices(interp_matrices, error, mesh.extents()));
```

Comment:
This will copy the `error` even if no interpolation is performed. Same for the `tensor[I]` in the line 245. Not sure if we care though..